File: src/test/scala/com/exasol/cloudetl/bucket/AlluxioBucketTest.scala
package com.exasol.cloudetl.bucket

class AlluxioBucketTest extends AbstractBucketTest {

  test("apply sets correct configuration") {
    val properties = Map(
      PATH -> "alluxio://alluxio/fs/path",
      FORMAT -> "parquet"
    )
    val exaMetadata = mockConnectionInfo("", "")
    val bucket = getBucket(properties, exaMetadata)
    val conf = bucket.getConfiguration()
    assert(bucket.isInstanceOf[AlluxioBucket])
    assert(conf.get("fs.alluxio.impl") === classOf[alluxio.hadoop.FileSystem].getName())
    assert(
      conf.get("fs.AbstractFileSystem.alluxio.impl") === classOf[alluxio.hadoop.AlluxioFileSystem]
        .getName()
    )
  }

}

File: src/test/java/com/exasol/cloudetl/bucket/AlluxioBucketTest.java
package com.exasol.cloudetl.bucket;

import static org.junit.jupiter.api.Assertions.*;

import java.util.Map;

import org.junit.jupiter.api.Test;

class AlluxioBucketTest extends AbstractBucketTest {
    @Test
    void createSetsCorrectConfiguration() {
        final Map<String, String> properties = Map.of(PATH, "alluxio://alluxio/fs/path", FORMAT, "parquet");
        final Bucket bucket = getBucket(properties, mockConnectionInfo("", ""));
        final org.apache.hadoop.conf.Configuration conf = bucket.getConfiguration();
        assertInstanceOf(AlluxioBucket.class, bucket);
        assertEquals(alluxio.hadoop.FileSystem.class.getName(), conf.get("fs.alluxio.impl"));
        assertEquals(alluxio.hadoop.AlluxioFileSystem.class.getName(),
                conf.get("fs.AbstractFileSystem.alluxio.impl"));
    }
}
